build/patch-prometheus.sh

```diff
@@ -100,6 +100,14 @@ function prometheus_patch_deployment() {
     rollingUpdate:\
       maxSurge: 25%\
       maxUnavailable: 25%' $file
+  
+  sed_wrap -i -e '/        - name: istio-proxy/ a\
+          resources:\
+{{- if .Values.global.proxy.resources }}\
+{{ toYaml .Values.global.proxy.resources | indent 12 }}\
+{{- else }}\
+{{ toYaml .Values.global.defaultResources | indent 12 }}\
```

Comment:
Should we verify that defaultResources is not empty before using it?

Reply:
shouldn't be necessary (and I blatantly copy-pasted this from other places where it's used exactly like this). it's always gonna have [something set](https://github.com/maistra/istio-operator/blob/d6c8a08b2cd839bd655bd8b8ae847a0cedfea2c1/resources/helm/v2.0/global.yaml#L354-L356)

Reply:
My concern would be that we end up with an empty resources block, which I'm not sure would be valid yaml or is interpreted as a null value.